core/deployment/src/main/java/io/quarkus/deployment/dev/IsolatedDevModeMain.java

```diff
@@ -126,6 +142,42 @@ public void accept(Integer integer) {
         }
     }
 
+    private void startCodeGenWatcher(QuarkusClassLoader classLoader, List<CodeGenData> codeGens) {
+        Executors.newSingleThreadExecutor().execute(
+                () -> {
+                    Collection<FSWatchUtil.Watcher> watchers = new ArrayList<>();
```

Comment:
Not a big fan of watchers, it's slightly different from the hot reload approach. @stuartwdouglas WDYT?

Reply:
it's actively watching for file changes. That's to enable users to use the changes right away in their code (without doing a request to the app first).

Reply:
We don't want to use the normal hot reload approach here. Otherwise you need to reload your browser for the generated source code to be generated, go back to your IDE to actually use the generated source, then back to your browser to see the changes.

Reply:
Make sense.